e2e/bitrise.yml

```diff
@@ -247,7 +425,7 @@ workflows:
         - verbose: "yes"
         - should_retry_test_on_fail: $RETRY_ON_FAILURE
         - cache_level: $CACHE_LEVEL
-        - collect_simulator_diagnostics: on_failure
+        - collect_simulator_diagnostics: $COLLECT_SIM_DIAGNOSTICS
```

Comment:
what is the reason behind this change?

Reply:
There are tests that we except to fail (flaky test retries for example), and in that case the diagnostics are collected even if is just  a test failure not a test runner failure (We do not have an up-to-date logic to know which one is which kind of failure)